### ISSUE TITLE
-XX:InitialRAMPercentage / -XX:MaxRAMPercentage  as an alternative to -Xms /-Xmx

### ISSUE THREAD

**Issue by ovlx** [MultiMC/MultiMC5#3995]
### Role

Modded minecraft/vanilla

### Suggestion

Allow for the use of the  InitialRAMPercentage and MaxRAMPercentage options

### Benefit

More convenient ,in particular for systems with limited memory.

### This suggestion is unique

- [X] I have searched the issue tracker and did not find an issue describing my suggestion, especially not one that has been rejected.

### You may use the editor below to elaborate further.

Currently Multimc always sets the Xmx and Xms flags, and adds them to the end of any user set java arguments, This prevents the usage of the  -XX:InitialRAMPercentage / -XX:MaxRAMPercentage options , since the Xms/Xmx options will always override them, 

Supporting these options directly, or an option to disable the Xms/Xmx flags  are both potential solutions

**Comment by embeddedt** [comment]
>  (guarantees the jvm will never allocate more memory than the system physically has.)

I have just read a bit about these options and I think this statement is incorrect. These options allow the *heap size* to be specified as a percentage, not the JVM's overall memory usage. As far as I can tell, other allocations/internal GC housekeeping structures are not considered.

**Comment by ovlx** [comment]
> > (guarantees the jvm will never allocate more memory than the system physically has.)
> 
> I have just read a bit about these options and I think this statement is incorrect, or at least misleading. These options allow the _heap size_ to be specified as a percentage, not the JVM's overall memory usage. As far as I can tell, other allocations/internal GC housekeeping structures are not considered.

That is true, however i still feel the overall point stands, if you set something like 100% , you aren't going to swapping nearly as bad as if you were to accidentally set 8gb on a 4gb system, may actually be acceptable if memory compression techniques are in use (supported on windows 10, and via zswap on linux)
